### ISSUE TITLE
Init signature gone for builtins

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#9616]
#9335 seem to have broken some things in inspect, `Init signature` is sometime gone:

```
(rootenv)~/dev/ipython master$ ipython -c 'list?'
Docstring:
list() -> new empty list
list(iterable) -> new list initialized from iterable's items
Type:      type

(rootenv)~/dev/ipython master$ git checkout 4.2.0
...

(rootenv)~/dev/ipython (4.2.0)$ ipython -c 'list?'
Init signature: list(self, /, *args, **kwargs)
Docstring:
list() -> new empty list
list(iterable) -> new list initialized from iterable's items
Type:           type
```

**Comment by Carreau** [comment]
ping @minrk

**Comment by minrk** [comment]
Fixed by #9617
